cmd/run.go

```diff
@@ -381,9 +381,8 @@ func initRuntime(ctx context.Context, params runCmdParams, args []string, addrSe
 	rt.SetDistributedTracingLogging()
 	rt.Params.AddrSetByUser = addrSetByUser
 
-	// v0 negates v1
-	if !addrSetByUser && !rt.Params.V0Compatible && rt.Params.V1Compatible {
-		rt.Params.Addrs = &[]string{defaultLocalAddr}
+	if !addrSetByUser && rt.Params.V0Compatible {
+		rt.Params.Addrs = &[]string{defaultAddr}
```

Comment:
I wonder, should we really default to `0.0.0.0` if the `--v0-compatible` flag is set? This has security implications. If you can add `--v0-compatible` to your command, you can also add `--addr ':8181'`.

Reply:
This is existing behavior which the flag is supposed to maintain. We already have a section on the security implications of this in the docs and we also log a warning on the command line. So I don't see a reason to change this.